File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    import math
    from collections import Counter
    
    rest = [100 - progresses[i] for i in range(len(progresses))]
    days = [math.ceil(rest[i] / speeds[i]) for i in range(len(rest))]
    
    for i in range(1, len(days)):
        if days[i] < days[i-1]:
            days[i] = days[i-1]
            
    return list(Counter(days).values())

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int size = progresses.length;

        int[] arr = new int[size];

        for (int i = 0; i < size; i++) {
            arr[i] = (100 - progresses[i] + speeds[i] - 1) / speeds[i];
        }

        for (int i = 1; i < size; i++) {
            if (arr[i] < arr[i - 1]) arr[i] = arr[i - 1];
        }

        LinkedList<Integer> q = new LinkedList<>();

        for (int i = 0; i < size; i++) {
            q.add(arr[i]);
        }

        List<Integer> ans = new ArrayList<>();

        while (!q.isEmpty()) {
            int cnt = 1;
            int cur = q.removeFirst();

            while (!q.isEmpty() && q.getFirst() == cur) {
                q.removeFirst();
                cnt++;
            }

            ans.add(cnt);
        }

        int[] answer = new int[ans.size()];
        for (int i = 0; i < ans.size(); i++) {
            answer[i] = ans.get(i);
        }

        return answer;
    }
}